Control Access section improvements (#2816)
sensu/sensu-docs: content/sensu-go/6.1/operations/control-access/_index.md

You can use Sensu's [built-in basic authentication provider][14] or configure [external authentication providers][15].

{{% notice note %}}
**NOTE**: For API-specific authentication, see the [API overview](../../api/#access-control) and [Use API keys to authenticate to Sensu](use-apikeys/).
{{% /notice %}}

Authorization establishes and manages user permissions: the extent of access users have for different Sensu resources.
Configure authorization with [role-based access control (RBAC)][4] to exercise fine-grained control over how they interact with Sensu resources.

## Authentication

Administrators can use Sensu's [built-in basic authentication provider][14] or configure [external authentication providers][15] to authenticate via Lightweight Directory Access Protocol (LDAP), Active Directory (AD), or OpenID Connect 1.0 protocol (OIDC).

### Use built-in basic authentication

Sensu's built-in basic authentication provider allows you to create and manage user credentials (usernames and passwords) with the [users API][53], either directly or using [sensuctl][2].
The basic authentication provider does not depend on external services and is not configurable.

> Sensu records basic authentication credentials in [etcd][54].

Reviewer: ```suggestion
Sensu hashes user passwords using the [bcrypt][26] algorithm and records the basic authentication credentials in [etcd][54].
```

link bcrypt to https://en.wikipedia.org/wiki/Bcrypt

* Microsoft AD, including Azure AD: [AD configuration examples][31] and [specification][32]
* OIDC tools like Okta and PingFederate: [OIDC configuration examples][9] and [specification][12]

**2. Apply the configuration with sensuctl**

Log in to sensuctl as the [default admin user][3] and apply the configuration to Sensu:

{{< code shell >}}
sensuctl create --file filename.json
{{< /code >}}

Use sensuctl to verify that your provider configuration was applied successfully:

{{< code shell >}}
sensuctl auth list

 Type     Name    
────── ────────── 
 ldap   openldap  
{{< /code >}}

#### Manage authentication providers

View and delete authentication providers with the [authentication providers API][27] or these sensuctl commands.

To view active authentication providers:

{{< code shell >}}
sensuctl auth list
{{< /code >}}

To view configuration details for an authentication provider named `openldap`:

{{< code shell >}}
sensuctl auth info openldap
{{< /code >}}

To delete an authentication provider named `openldap`:

{{< code shell >}}
sensuctl auth delete openldap
{{< /code >}}

> 

Reviewer: Consider adding a high-level example of logging in via sensuctl to complete the circle. This gives is a place to add a note similar to `Users _do not_ need to provide the username_prefix when logging in to Sensu.` which is currently only documented for individual authentication providers.

## Authorization

After you set up authentication, configure authorization via [role-based access control (RBAC)][4] to give those users permissions within Sensu.
Sensu RBAC allows you to manage and access users and resources based on namespaces, groups, roles, and bindings.
See [Create a read-only user][5] for an example.

- **Namespaces** partition resources within Sensu.
Sensu entities, checks, handlers, and other [namespaced resources][17] belong to a single namespace.
- **Roles** create sets of permissions (like GET and DELETE) tied to resource types.

> **Cluster roles** apply permissions across namespaces and include access to [cluster-wide resources][18] like users and namespaces. 

Reviewer: ```suggestion
**Cluster roles** apply permissions across all namespaces and may include access to [cluster-wide resources][18] like users and namespaces. 
```

- **Role bindings** assign a role to a set of users and groups within a namespace.

> **Cluster role bindings** assign a cluster role to a set of users and groups cluster-wide.

Reviewer: ```suggestion
**Cluster role bindings** assign a cluster role to a set of users and groups across all namespaces.
```

To enable permissions for external users and groups within Sensu, you can create a set of [roles, cluster roles][11], [role bindings, and cluster role bindings][13] that map to the usernames and group names in your authentication provider.

> After you configure the correct roles and bindings, users can log in to [sensuctl][36] and the [web UI][1] using a single-sign-on username and password (no prefix required).

Reviewer: ```suggestion
After you configure an authentication provider and establish the roles and bindings to grant authenticated users the desired privileges, those users can log in via [sensuctl][36] and the [web UI][1] using a single-sign-on username and password (no prefix required).
```

[1]: ../../web-ui/
[2]: ../../sensuctl/
[3]: rbac#default-users
[4]: rbac/
[5]: create-read-only-user/
[6]: ../../commercial/
[7]: oidc-auth/
[8]: ../../api/
[9]: oidc-auth/#oidc-configuration-examples
[10]: https://docs.microsoft.com/en-us/azure/active-directory-domain-services/tutorial-configure-ldaps
[11]: rbac#roles-and-cluster-roles
[12]: oidc-auth/#oidc-specification
[13]: rbac#role-bindings-and-cluster-role-bindings
[14]: #use-built-in-basic-authentication
[15]: #use-an-authentication-provider
[17]: rbac#namespaced-resource-types
[18]: rbac#cluster-wide-resource-types
[19]: ../maintain-sensu/troubleshoot#log-levels
[21]: ldap-auth/#ldap-group-search-attributes
[22]: ldap-auth/#ldap-user-search-attributes
[23]: ad-auth/#ad-metadata-attributes
[24]: ldap-auth/#ldap-metadata-attributes
[25]: /oidc-auth/#oidc-spec-attributes

> [27]: ../../api/authproviders/

Author: ```suggestion
[26]: https://en.wikipedia.org/wiki/Bcrypt
[27]: ../../api/authproviders/
```
